apollo-router/src/services/execution/service.rs

```diff
@@ -321,8 +339,32 @@ impl ExecutionService {
                     schema.api_schema(),
                     variables_set,
                 );
+
+                if let (ApolloMetricsReferenceMode::Extended, Some(Value::Object(response_body))) = (metrics_ref_mode, &response.data) {
+                    extract_enums_from_response(
+                        filtered_query.clone(),
+                        operation_name,
+                        schema.api_schema(),
+                        response_body,
+                        &mut referenced_enums,
+                    );
+                }
+            }
+
+            if let (ApolloMetricsReferenceMode::Extended, Some(Value::Object(response_body))) = (metrics_ref_mode, &response.data) {
+                extract_enums_from_response(
+                    query.clone(),
+                    operation_name,
+                    schema.api_schema(),
+                    response_body,
+                    &mut referenced_enums,
+                );
```

Comment:
this should not be done twice. `format_response` happens twice because it first has to go through the filtered query (with fields removed due to authorization), then the client query, to get the expected shape by inserting nulls were appropriate. `extract_enums_from_response` should happen only once, and after those steps

Reply:
So does the `query` contain all the enums that are also in `filtered_query` at this point in the code?

Reply:
if an enum is annotated with `@authenticated` or other authorization directives and the client is not authorized, then the filtered query will not have the fields of that enum type. Do you want to measure what is obtained in query plan execution, or what is returned to the client?

Reply:
The reason for returning the enum usage is mainly so we can see if it's safe to remove an unused enum, so I think it's better to use the query plan execution.